src/python/pants/engine/build_files.py

```diff
@@ -319,27 +319,12 @@ def spec_to_globs(address_mapper, specs):
     elif type(spec) is AscendantAddresses:
       patterns.update(join(f, pattern)
                       for pattern in address_mapper.build_patterns
-                      for f in _recursive_dirname(spec.directory))
+                      for f in recursive_dirname(spec.directory))
```

Comment:
Separate change?

Reply:
The method is now used in file invalidation in `test_base`.